ilc/client/GuardManager.js

```diff
@@ -0,0 +1,112 @@
+const navigateToUrl = require('single-spa').navigateToUrl;
+const unlocalizeUrl = require('../common/i18n/unlocalizeUrl');
+
+const actionTypes = require('../common/guard/actionTypes');
+
+const getIlcRouter = require('./getIlcRouter');
+const getIlcConfig = require('./ilcConfig');
+
+class GuardManager {
+    #registryConf;
+    #router;
+    #unlocalizeUrl;
+    #navigateToUrl;
+    #transitionHooksPlugin;
+
+    constructor(
+        registryConf,
+        router,
+        unlocalizeUrl,
+        navigateToUrl,
+        transitionHooksPlugin,
+    ) {
+        this.#registryConf = registryConf;
+        this.#router = router;
+        this.#unlocalizeUrl = unlocalizeUrl;
+        this.#navigateToUrl = navigateToUrl;
+        this.#transitionHooksPlugin = transitionHooksPlugin;
+    }
+
+    shouldNavigateTo(url) {
+        const pathname = new URL(url, 'http://hack').pathname;
+        const route = this.#router.match(this.#unlocalizeUrl(this.#registryConf.settings.i18n, pathname));
+        const hooks = this.#transitionHooksPlugin.getTransitionHooks();
+
+        if (route.specialRole == null) {
+            for (const hook of hooks) {
+                const action = hook({
+                    route: {
+                        ...route,
+                        url,
+                    },
+                    navigate: this.#navigateToUrl,
```

Comment:
I am not positive that we have to provide just navigateToUrl method from single-spa
Probably, we have to check that an URL that would be provided from a hook single-spa has